src/test/scala/tech/cryptonomic/conseil/tezos/DataTypesTest.scala

```diff
@@ -72,22 +86,23 @@ import scala.concurrent.ExecutionContext.Implicits.global
         aggregation = None
       )
 
-      val result = query.validate("test", tpdo)
+      val result = query.validate(testEntityPath, ms)
 
       result.futureValue.right.get shouldBe Query(predicates = List(Predicate("valid", OperationType.in)))
     }
 
     "return error with incorrect predicate fields" in {
       val attribute = Attribute(
-        name = "valid",
-        displayName = "valid",
+        name = "invalid",
+        displayName = "invalid",
         dataType = DataType.Int,
         cardinality = None,
-        keyType = KeyType.NonKey,
+        keyType = KeyType.UniqueKey,
         entity = "test"
       )
-      (tpdo.isAttributeValid _).when("test", "invalid").returns(Future.successful(false))
-      (tpdo.getTableAttributesWithoutUpdatingCache _).when("test").returns(Future.successful(Some(List(attribute))))
+      (ms.isAttributeValid _).when("testEntity", "invalid").returns(Future.successful(false))
```

Comment:
Maybe not for now but I would consider replacing those heavy stubbing with some real implementation dedicated for tests when we can provide a configuration in a more clean way.